.github/workflows/ci.yml

```diff
@@ -14,10 +14,48 @@ concurrency:
 
 jobs:
 
+  changes:
+    name: "Detect changes"
+    runs-on: "ubuntu-latest"
+
+    outputs:
+      pkgdb: "${{ steps.changes.outputs.pkgdb }}"
+      env-builder: "${{ steps.changes.outputs.env-builder }}"
+      cli: "${{ steps.changes.outputs.cli }}"
+
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
```

Comment:
@aakropotkin I'm not 100% sure about the paths which should trigger the rebuilds I'm sure you will know better. I guess you also need to first finish the top-level Makefile work, but I went ahead and create this PR anyway.

Reply:
Yeah wait for the top level Makefile. I plan to PR it tomorrow.

It's basically "anything under the subsystem name redoes the subsystem and consumers".

There's some magic top level files that trigger everything.

There's some boring files at the top level that are cosmetic/functionally irrelevant.